app/builders/v2/report_builder.rb

```diff
@@ -56,6 +58,10 @@ def label
     @label ||= account.labels.where(id: params[:id]).first
   end
 
+  def team
+    @team ||= account.teams.where(id: params[:id]).first
```

Comment:
`.find` is enough right?

Reply:
`.find` would throw an exception if record not found but `.where` return an empty array.

Reply:
In that case use `find_by(id: params[:id])`.

Reply:
Is there any specific reason for `where` condition? Because i can see the same for inbox, user and label above.

@sojan-official

Reply:
`find_by` uses `where(id: params[:id]).take` internally which is faster than `where(id: params[:id]).first` so as we need first matching record always go with find or find_by according to if you want to handle the exception or not.

Reply:
@aswindevps  Using .find isn't a bad choice here.  The reason why .find would throw and exception is because the param supplied to the method is an invalid team id.  ( wrong id format or a team id that doesn't belong to the account )
In such cases we would want the error to be thrown, which will be handled by https://github.com/chatwoot/chatwoot/blob/0c24df96a8444b7e85b0a8395deb9c34371767aa/app/controllers/concerns/request_exception_handler.rb#L12 and we would throw a 404 .

we could switch to find for labels and inbox as well.

Reply:
Switched to `.find` method for inboxes, agents, labels and teams.